src/middleware/navigation/has.presenter.middleware.ts

```diff
@@ -1,18 +1,8 @@
 import { Request, Response, NextFunction } from 'express';
 
-import { logger } from '../../utils/logger';
+import { checkIfPresenterDataExists } from './common-code/has.presenter';
 import { SOLD_LAND_FILTER_URL } from '../../config';
-import { getApplicationData } from "../../utils/application.data";
-import { checkPresenterDetailsEntered, NavigationErrorMessage } from './check.condition';
 
 export const hasPresenter = (req: Request, res: Response, next: NextFunction): void => {
-  try {
-    if ( !checkPresenterDetailsEntered(getApplicationData(req.session)) ) {
-      logger.infoRequest(req, NavigationErrorMessage);
-      return res.redirect(SOLD_LAND_FILTER_URL);
-    }
-    next();
-  } catch (err) {
-    next(err);
-  }
+  checkIfPresenterDataExists(req, res, next, SOLD_LAND_FILTER_URL);
```

Comment:
@olaych @stephanie-ch 

does this seem like a reasonable way to split out shared code between the new navigation check for Presenter and the existing one in the register journey?

Also I put it in navigation/common-code for now, not sure where its home should be or what it should be called?

Reply:
@AndreKramer7654  any thoughts on where this shared code for navigation checks should be kept? navigation/common-code is temporary until we find a home for it

Reply:
navigation/checks ?

Reply:
updated to navigation/checks